### PR TITLE
Foundations/ Revisiting RPS: Add lesson overview and knowledge check

### LINKED ISSUE #27332: Foundations/Revisiting Rock Paper Scissors: Update Lesson Structure
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

The "Revisiting RPS" lesson currently lacks a conventional lesson structure, as it is not a project but rather a lesson. To align with the standard lesson format and ensure consistency across the curriculum, it should include a "lesson overview" section.

Additionally, including a "lesson overview" section will maintain consistency across the curriculum, ensuring that all lessons follow a standardised structure for improved user experience and clarity.

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-revisiting-rock-paper-scissors

### (Optional) Discord Name

_No response_

### (Optional) Additional Comments

This suggestion arose from the discussion in PR #27294, as suggested by @thatblindgeye.

### PR DESCRIPTION
**Pull request by maahsnd**
## Because

The "Revisiting RPS" lesson lacks both a "lesson overview" and a "knowledge check." Add both to align with standard lesson format. This will ensure consistency across the curriculum. Standardized structure improves clarity and user experience.

## This PR

- Add a lesson overview section. The lesson covers how to use git branches (create, merge, delete, best practices, use cases).
- Add a knowledge check section. Reference the general concept of branches, branch creation, branch merging, and one use case.

## Issue

Closes #27332

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: foundations/javascript_basics/revisiting_rock_paper_scissors.md [hunk 1]
```diff
@@ -22,19 +30,20 @@ Once you are done working on your feature branch and are ready to bring the comm
 
 Merges are done by using the command `git merge <branch_name>` which will take the changes you’ve committed in `branch_name` and add them to the branch that you’re currently on. You can see an example of a `develop` branch being created, committed to, and then merged to `main` in the diagram below.
 
-<pre class="mermaid">
----
-title: Example of Git Branching
----
+## <pre class="mermaid">
```
Comment (KevinMulhern) [inline foundations/javascript_basics/revisiting_rock_paper_scissors.md:33]: Nit: This is used to display a diagram, and needs to be in a very specific format to display correctly.
Can everything between this line and line 46 be reverted back to what it was before please?

**Review by KevinMulhern — CHANGES_REQUESTED** [review]

**Comment by maahsnd** [comment]
Should I make those changes/ revert the requested section? I did not intend to alter anything there, must have been in the linting process and I missed it. Sorry for the confusion and any hassle.

**Comment by KevinMulhern** [comment]
> Should I make those changes/ revert the requested section?

Yes please @maahsnd, we'll need those changes reverted to merge this PR.

**Comment by maahsnd** [comment]
Hey @KevinMulhern I attempted to make the change. In my local environment, the diagram code matches its earlier state. I've committed and pushed that to the branch. I'm not confident I did that correctly. I'm inexperienced with updating PRs and am concerned I made a mistake.

**Comment by KevinMulhern** [comment]
It looks like an empty merge commit was added, are you sure the revert changes were committed?

You can view the changes for this PR in the "Files changed" tab near the top of this page to verify whats in this PR.

**Comment by maahsnd** [comment]
Revert changes were not committed. I've made a mess of my branch/ local git, and am not sure how to clean that up. My thought is to start from scratch, opening a new PR.

**Comment by MaoShizhong** [comment]
@maahsnd If you've ended up in a bit of a tangle and you feel the easiest way for you to resolve this would be a fresh PR, feel free to close this one then open a new PR once you've done what you need to do.

**Comment by maahsnd** [comment]
@MaoShizhong Got it, I'll do that. Thank you.

**Comment by maahsnd** [comment]
@MaoShizhong @KevinMulhern I opened a new PR, #27810

**Pull request closed without merging**